application_test.go

```diff
@@ -450,17 +450,60 @@ func TestApplicationConfiguration(t *testing.T) {
 	assert.Equal(t, ErrCodeNotFound, apiErr.ErrCode)
 }
 
-func TestWaitOnApplication(t *testing.T) {
+func TestWaitOnApplicationWithTimeoutGreaterThanTicker(t *testing.T) {
+	defaultConfig := NewDefaultConfig()
+	configs := &configContainer{}
+	configs.client = &defaultConfig
+	configs.client.WaitTime = 100
```

Comment:
I find it more readable to set `WaitTime` right after `defaultConfig` has been defined. Moreover, I would use literal structs to be slightly less repetitive; so overall:

```
defaultConfig := NewDefaultConfig()
defaultConfig.WaitTime = 100
configs := &configContainer{
  client: &defaultConfig,
}
```